io_ogre/ogre/material.py

```diff
@@ -175,11 +202,57 @@ def generate_pass( self, mat, pass_name="" ):
                         if node.texture:
                             geo = shader.get_connected_input_nodes( self.material, node )[0]
                             self.generate_texture_unit( node.texture, name=name, uv_layer=geo.uv_layer )
-            elif images:
-                for image in images:
-                    self.generate_texture_unit(image)
+            elif textures:
+                for key, texture in textures.items():
+                    self.generate_image_texture_unit(key, texture)
 
+    def generate_image_texture_unit(self, key, texture):
+        """
+        Generates a texture_unit of a pass.
+        
+        key: key of the texture in the material wrapper (not used, for normal map if needed)
+        texture: the texture wrapper (node_shader_utils.PrincipledBSDFWrapper)
+        """
+        src_dir = os.path.dirname(bpy.data.filepath)
+        # For target path relative
+        # dst_dir = os.path.dirname(self.target_path)
+        dst_dir = src_dir
+        filename = io_utils.path_reference(texture.image.filepath, src_dir, dst_dir, mode='RELATIVE', library=texture.image.library)
+        # Do not use if target path relative
+        # filename = repr(filepath)[1:-1]
+        _, filename = split(filename)
+        filename = self.change_ext(filename, texture.image)
+        with self.w.iword('texture_unit').embed():
+            self.w.iword('texture').word(filename).nl()
+            
+            exmode = texture.extension
+            if exmode in TEXTURE_ADDRESS_MODE:
+                self.w.iword('tex_address_mode').word(TEXTURE_ADDRESS_MODE[exmode]).nl()
 
+            if exmode == 'CLIP':
+                self.w.iword('tex_border_colour').word(texture.color.r).word(texture.color.g).word(texture.color.b).nl()
+            if texture.scale != Vector((1.0, 1.0, 1.0)):
+                self.w.iword('scale').real(1.0 / texture.scale.x).real(1.0 / texture.scale.y).nl()
+            
+            if texture.texcoords == 'Reflection':
+                if texture.projection == 'SPHERE':
+                    self.w.iline('env_map spherical')
+                elif texture.projection == 'FLAT':
+                    self.w.iline('env_map planar')
+                else: 
+                    logging.debug('WARNING: <%s> has a non-UV mapping type (%s) and not picked a proper projection type of: Sphere or Flat' %(texture.name, texture.projection))
+            
+            if texture.translation.x or texture.translation.y:
+                self.w.iword('scroll').real(texture.translation.x).real(texture.translation.y).nl()
+            if texture.rotation.z:
```

Comment:
it seems blender never actually supported texture rotation:
https://blender.stackexchange.com/questions/5608/rotate-object-texture

and somebody just picked a free variable for whatever reasons

Reply:
These values (location, rotation, scale) are retrieved from a Vector Mapping node.
That's how I tested it, and I guess the only way to change an image texture mapping on Blender 2.8x.
What is the unit for ogre rotation ?